### PR TITLE
python312Packages.pyopengltk: init at 0.0.4

### LINKED ISSUE #336341: Package request: `python3Packages.pyopengltk`
**Project description**

allows you to draw to an OpenGL canvas in a tk(inter) window in python 3

**Metadata**

* homepage URL: https://pypi.org/project/pyopengltk/
* source URL: https://github.com/jonwright/pyopengltk
* license: mit
* platforms: linux, darwin, win32 (possibly others)

**Additional information**

i'd package this myself but i don't know how to

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by Sigmanificient**
## Description of changes

- #336341

Fixes: #336341

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Sigmanificient** [comment]
@ofborg eval python312Packages.pyopengltk.passthru.tests

**Comment by Sigmanificient** [comment]
```
$ nix-build -A python3Packages.pyopengltk.passthru.tests
/nix/store/lxgj4j0lz9dy0m8j229frqr3r5gk5ann-cube
$ result/bin/cube
```

image

File: pkgs/development/python-modules/pyopengltk/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    hash = "sha256-hQoTj8h/L5VZgmq7qgRImLBKZMecrilyir5Ar6ne4S0=";
+  };
+
+  build-system = [ setuptools ];
+
+  dependencies = [
+    pyopengl
+    tkinter
+  ];
+
```
Comment (natsukium) [inline pkgs/development/python-modules/pyopengltk/default.nix:30]: Could you enable `pythonImportsCheck`?

You can see a checklist in the contribution guide.
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#contributing-guidelines-contributing-guidelines

File: pkgs/development/python-modules/pyopengltk/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  passthru.tests = {
+    cube = writers.writePython3 "cube" {
+      libraries = [ pyopengltk ];
+      doCheck = false;
+    } (builtins.readFile "${src}/examples/cube.py");
+  };
+
+  meta = {
+    description = "OpenGL frame for Python/Tkinter via ctypes and pyopengl";
```
Comment (natsukium) [inline pkgs/development/python-modules/pyopengltk/default.nix:41]: Please add `meta.homepage`.

**Review by natsukium — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pyopengltk/default.nix (hunk 1 above)
Comment (Sigmanificient) [reply pkgs/development/python-modules/pyopengltk/default.nix:30]: I keep forgetting that one everytime

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 336360` at de07bc4b run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/natsukium/6b9d6bd92b70a76c42ed4a13d96bc146">python311Packages.pyopengltk</a></li>
    <li><a href="https://gist.github.com/natsukium/6829d0848afb85fd7956319fd3225234">python312Packages.pyopengltk</a></li>
  </ul>
</details>

**Comment by Sigmanificient** [comment]
I dont have a `aarch64-darwin` to test :<

**Comment by natsukium** [comment]
It doesn't seem to support darwin yet. https://github.com/jonwright/pyopengltk/issues/12

**Comment by Sigmanificient** [comment]
so `broken = stdenv.isDarwin`?

**Comment by natsukium** [comment]
Yes, please.

**Review by natsukium — APPROVED** [review]

**Pull request merged**